File: Data Structure I/cpp/Day 6/FirstUniqueChar.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        /*
        Approach
          1- Count char frequency in s
          2- return first with count 1
        */
        int map[26+1]={};
        for(int i=0;i<s.length();i++){
            map[s[i]-96]++;
            
        }
        for(int i=0;i<s.length();i++){
            if(map[s[i]-96]==1){
                return i;
            }
        }
        return -1; 
    }
};

File: Data Structure I/python/Day 6/first_unique_char.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        '''
        Approach
          1- Count char frequency in s
          2- return first with count 1
        '''
        map={}
        for i in range(len(s)):
            if s[i] in map:
               map[s[i]]+=1
            else:
                map[s[i]]=1
        for i in range(len(s)):
            if(map[s[i]]==1):
                return i
        return -1; 